File: backend/src/app/Models/ProviderCoilMovement.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class ProviderCoilMovement extends Model
{
    use HasFactory;

    protected $fillable = [
        'provider_id',
        'raw_material_entry_id',
        'type',
        'coils_received',
        'coils_returned',
    ];

    public function provider()
    {
        return $this->belongsTo(Provider::class);
    }

    public function entry()
    {
        return $this->belongsTo(RawMaterialEntry::class, 'raw_material_entry_id');
    }
}

File: backend-dotnet/src/Ikhana.Domain/Entities/ProviderCoilMovement.cs
using Ikhana.Domain.Enums;

namespace Ikhana.Domain.Entities;

public class ProviderCoilMovement : BaseEntity
{
    public long ProviderId { get; set; }
    public long? RawMaterialEntryId { get; set; }

    public MovementType Type { get; set; } = MovementType.Entry;

    public int CoilsReceived { get; set; }
    public int CoilsReturned { get; set; }

    public Provider Provider { get; set; } = null!;
    public RawMaterialEntry? Entry { get; set; }
}
